Add reset password

aespasa opened the pull request:
close #25

aespasa committed:
Add reset password

aespasa committed:
Remove double password verification when register

aespasa committed:
Increase reset token security

aespasa committed:
Add isLoggedIn and isAdmin rules to controllers. WIP on WYGIWYS editor and Dashboard

TBoileau commented on config/defines.php:
```diff
@@ -0,0 +1,3 @@
+<?php
+
+define('_ROOT_', (!empty($_SERVER['HTTPS']) ? 'https' : 'http') . '://' . $_SERVER['HTTP_HOST']);
```
Non

TBoileau commented on src/Controller/SecurityController.php:
```diff
@@ -73,53 +82,171 @@ public function login(
+                    ->priority(Email::PRIORITY_NORMAL)
+                    ->subject("Réinitialisation de votre mot de passe")
+                    ->text(strip_tags($messageTxt))
+                    ->html(nl2br($messageHtml));
```
Implémenter une méthode `renderView` dans `AbstractController` qui n'aura pour seule responsabilité, de renvoyer la vue sous format string.

TBoileau commented on src/Controller/SecurityController.php:
```diff
@@ -73,53 +82,171 @@ public function login(
-                    "Votre demande de création de compte a été prise en compte, vous recevrez un mail lorsque celle-ci sera validée"
-                );
+                // @phpstan-ignore-next-line
+                $resetUrl = _ROOT_ . $this->router->generateUri('handle_reset_password');
```
Utiliser la `$request`, notamment les méthode `getScheme` et `getHost`

TBoileau commented on src/Controller/SecurityController.php:
```diff
@@ -73,53 +82,171 @@ public function login(
+        UserRepository $userRepository,
+        Authenticator $auth
+    ): Response {
+        if ($auth->isLoggedIn()) {
+            return $this->redirect('/');
+        }
```
Implémenter une méthode `denyAccessUnlessLogged` dans `AbstractController`, qui se chargera de vérifier si l'utilisateur est connecté, sinon throw une erreur ou de faire comme tu le fais une redirection vers la page de connexion

TBoileau commented on src/Controller/DashboardController.php:
```diff
@@ -5,24 +5,59 @@
+            dd($post);
+        }
+
+        return $this->render('pages/back/form_post.html.twig', ['form' => $form, 'errors' => $form->getErrors()]);
```
Tu pourrais simplement faire `form.errors` dans twig au lieu de passer les erreurs

aespasa committed:
Applying changes

Replace Password Strength constraint validator because it was not what we expected at first.

Reset Password feature: Add a renderView method inside the abstract controller to load a view without returning something. In this way we can load the new mails/* views.

There was an unexpected behavior from the Templating and the Abstract Controller method about the flashes messages. Now, we specify if we are in the main request in case we want to call something else (e.g renderView()) without clearing the flashBag from the session.

Add error 500 handling and Unauthorized exception redirecting the user to the login page.

aespasa committed:
Refactor errors
